src/modules/battery.rs

```diff
@@ -19,43 +19,79 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     let battery_status = get_battery_status()?;
     let BatteryStatus { state, percentage } = battery_status;
 
-    if percentage > BATTERY_THRESHOLD {
-        log::debug!(
-            "Battery percentage is higher than threshold ({} > {})",
-            percentage,
-            BATTERY_THRESHOLD
-        );
-        return None;
-    }
-
-    // TODO: Set style based on percentage when threshold is modifiable
     let mut module = context.new_module("battery");
-    let module_style = module
-        .config_value_style("style")
-        .unwrap_or_else(|| Color::Red.bold());
-    module.set_style(module_style);
-    module.get_prefix().set_value("");
-
-    match state {
-        battery::State::Full => {
-            module.new_segment("full_symbol", BATTERY_FULL);
-        }
-        battery::State::Charging => {
-            module.new_segment("charging_symbol", BATTERY_CHARGING);
-        }
-        battery::State::Discharging => {
-            module.new_segment("discharging_symbol", BATTERY_DISCHARGING);
+
+    // Parse config under `display`
+    let display_styles = get_display_styles(&module);
+    let display_style = display_styles.iter().find(|display_style| {
+        let BatteryDisplayStyle { threshold, .. } = display_style;
+        percentage <= *threshold as f32
+    });
+
+    if let Some(display_style) = display_style {
+        let BatteryDisplayStyle { style, .. } = display_style;
```

Comment:
Is `style` here the first element of the struct because `threshold` was already destructured out earlier?